src/Sets/Polygon.jl

```diff
@@ -0,0 +1,38 @@
+export Polygon
+
+"""
+    Polygon{N, VN<:AbstractVector{N}} <: AbstractPolygon{N}
+
+Type that represents a polygon by its line segments.
+
+### Fields
+
+- `line_segments` -- the list of line segments
+
+### Notes
+
+This type assumes that all line segments are sorted in counter-clockwise fashion.
+
+To ensure this property, the constructor of `Polygon` runs a non-convex-hull
+algorithm on the line segments by default. This also removes redundant line segments.
+If the line segments are known to be sorted, the flag `apply_convex_hull=false` can
+be used to skip this preprocessing.
+
+### Examples
+
+```
+"""
+struct Polygon{N, VN<:AbstractVector{N}} <: AbstractPolygon{N}
+    line_segments::LineSegment{N, VN<:AbstractVector{N}}
```

Comment:
```suggestion
    edges::Vector{LineSegment{N, VN}}
```
we could in principle use any outer array type, but `Vector` is also fine.